lib/statesman/adapters/active_record.rb

```diff
@@ -184,15 +284,45 @@ def with_updated_timestamp(params)
                    ActiveRecordTransition::DEFAULT_UPDATED_TIMESTAMP_COLUMN
                  end
 
-        return params if column.nil?
+        # No updated timestamp column, don't return anything
+        return nil if column.nil?
 
-        timestamp = if ::ActiveRecord::Base.default_timezone == :utc
-                      Time.now.utc
-                    else
-                      Time.now
-                    end
+        [
+          column, ::ActiveRecord::Base.default_timezone == :utc ? Time.now.utc : Time.now
+        ]
+      end
+
+      def db_mysql?
+        ::ActiveRecord::Base.connection.adapter_name.downcase.starts_with?("mysql")
```

Comment:
Is there any way we could avoid special casing this for mysql?

If that's not possible could we make this a configuration option instead? Selfish reasoning: we use an internal adapter that is not called `mysql`. I can monkey patch this method internally, but if there is any way to avoid it that would be sweet.

Reply:
I think that's a much better approach actually, makes this change opt-in which is great. Will try find time later today.

Reply:
On the not making it a special case, I did try implementing a platform agnostic strategy but the requirements of Postgres and MySQL were too difficult to get them to marry. Where MySQL could do X, Postgres would suffer, and vise versa.

Agree with what you're saying though, providing a configuration option for this would be best. We can even default it to true if you're MySQL, while giving you everything you need.

Thanks for the prompt!

Reply:
Yeah that sounds great, and the explanation makes sense 👍 

Scope creep: at this point, should we consider creating separate `Statesman::Adapter`s (one for postgres, one for mysql) for the different AR adapters? That way we can avoid some of the branching that we see creeping in here.

Reply:
That might be a good idea, though if we're allowing this to be configurable, I guess you'd want something more like a mixin. `adapter.include(Statesman::Adapters::MySQLGapLock` or something.

That said, let's leave that until after we've got this in. I'm conscious of how long this has been hanging, very keen to deliver the value then clean-up!

Reply:
Managed to get this passing this morning, took way longer than it should have as you can see with all the force pushes 🙈 lemme know what you think. I think config may need a small refactor next time we add one of these configuration options.

Reply:
I think this looks great, thanks so much 🎉 ❤️